### DISCUSSION TITLE
partial modifier is missing on CustomControl

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#10563] [Q&A] [answered]
I am following [these instructions](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html) to create a control library.

I added a Custom Control and the generated class looks like this:

    public sealed class CustomControl1 : Control
	
which causes this error:

> error CS0260: Missing partial modifier on declaration of type 'CustomControl1'; another partial declaration of this type exists

Other than `Class1` which is supplied by default when the project is created, there are no other controls or classes in the project.

Adding the `partial` modifier allows the project to build.

Interesting to note:  The error message does not appear in the VS Error List window.  I had to go to compliler output window to find it.  I don't recall ever getting an error in VS that does not show up in the error list window.

See also [this question](https://github.com/unoplatform/uno/discussions/10562).

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. This is an expected behavior, as all `UIElement` inheriting classes contain generated code which need the partial modified. 

That being said, the template should contain the partial modifier by default. Let's track this in https://github.com/unoplatform/uno/issues/10565.
